sources/en/w/wuxiacom.py

```diff
@@ -19,17 +20,54 @@ class WuxiaComCrawler(BasicBrowserTemplate):
     ]
 
     def initialize(self):
-        # self.headless = True
+        #self.headless = True
         self.api_url = "https://api2.wuxiaworld.com"
         self.grpc = RpcSession.from_descriptor(WUXIWORLD_PROTO)
         self.grpc._session = self.scraper
         self.cleaner.bad_css.clear()
         self.cleaner.bad_tags.add("hr")
         self.bearer_token = None
         self.cleaner.unchanged_tags.update(["span"])
+        self.wuxia_login_info = list()
+        self.start_download_chapter_body_in_browser = False
+        #self.loged_in = False
 
     def login(self, email: str, password: str) -> None:
-        self.bearer_token = email + " " + password
+        #self.loged_in = False
+        if email == 'Bearer':
+            logger.debug("login type: %s", email)
+            self.bearer_token = email + " " + password
+        else:
+            logger.debug("login type: Email(%s)", email)
+            self.init_browser()
+            self.visit('https://www.wuxiaworld.com/manage/profile/')
+            self.browser.wait("h6 button")
+            self.browser.find("h6 button").click()
+            self.browser.wait("input#Username")
+            self.browser.find("input#Username").send_keys(email)
+            self.browser.find("input#Password").send_keys(password)
+            self.browser.find("button").click()
+            #time.sleep(10) # wait 10sec instead for waiting for elemnt in case login failed
+            #driver.find_element(By.CSS_SELECTOR, "h6 button").click()
+            #driver.find_element(By.CSS_SELECTOR, "input#Username").send_keys(email)
+            #driver.find_element(By.ID, "Password").send_keys(password)
+            #driver.find_element(By.NAME, "button").click()
+            #self.wuxia_cookies = self.get_cookies()
+            #self._restore_cookies()
+            #logger.debug("cookies: Email(%s)", str(self.cookies))
+            #logger.debug("headers: Email(%s)", str(self.headers))
+            try:
```

Comment:
please remove the commented out codes. this helps the codebase to stay clean.

Reply:
sure
but what about headless? should i turn it on?

Reply:
no need. it will be automatically turned off for environments with no display